src/odf_ooxml_parser.cpp

```diff
@@ -120,9 +120,15 @@ class ODFOOXMLParser::CommandHandlersSet
 			ODFOOXMLParser& p = (ODFOOXMLParser&)parser;
 			int expected_col_num = p.lastOOXMLColNum() + 1;
 			std::string cell_addr_attr = xml_stream.attribute("r");
-			static const std::regex cell_addr_rgx("([A-Z]+)([0-9]+)");
+			bool matched;
 			std::smatch match;
-			if (std::regex_match(cell_addr_attr, match, cell_addr_rgx) && match.size() == 3)
+			static std::mutex cell_addr_rgx_mutex;
+			{
+				std::lock_guard<std::mutex> cell_addr_rgx_mutex_lock(cell_addr_rgx_mutex);
+				static const std::regex cell_addr_rgx("([A-Z]+)([0-9]+)");
+				matched = std::regex_search(cell_addr_attr, match, cell_addr_rgx);
+			}
```

Comment:
I think that according to C++ specification local static variables are initialized only once when the line is executed first time. It can be optimized like you said if multiple functions use same variable but it's not the case I think.